### PR TITLE
Adding command line arguments for Kotlin examples

### PR DESCRIPTION
**Pull request by DariusIMP**
Using Clickt to allow users to specify arguments when running examples on Kotlin.

# Usage

For instance the ZPub example, you run it using gradle, and providing the arguments the following way:

```bash
gradle ZPub --args="<your arguments>"
```

** Example: **
```bash
gradle ZPub --args="-h"
```

will display the help menu with the available arguments:
```
> Task :examples:ZPub
Usage: zpub [<options>]

  Zenoh Pub example

Options:
  -e, --connect=<connect>  Endpoints to connect to.
  -c, --config=<config>    A configuration file.
  -k, --key=<key>          The key expression to write to [default:
                           demo/example/zenoh-kotlin-pub]
  -l, --listen=<listen>    Endpoints to listen on.
  -m, --mode=<mode>        The session mode. Default: peer. Possible values:
                           [peer, client, router]
  -v, --value=<value>      The value to write. [Default: "Pub from Kotlin!"]
  -a, --attach=<attach>    The attachments to add to each put. The key-value
                           pairs are &-separated, and = serves as the separator
                           between key and value.
  --no-multicast-scouting  Disable the multicast-based scouting mechanism.
  -h, --help               Show this message and exit
```

Solves #55

### CODE REVIEW

**Comment by oteffahi** [comment]
`gradle ZPub --args="-l tcp/127.0.0.1:7447 -l udp/127.0.0.1:7447"` will only listen on UDP port.

As it is, a repeated flag will overwrite itself, so only the last instance counts. For list options (`-l`, `-e`), it would be good if we can repeat the flag for each value, as some people are already used to this syntax on Zenoh Rust examples (which support both syntaxes). If we dont support it here, repeating a flag multiple times should error-out to avoid unintended configuration.

File: examples/src/main/kotlin/io.zenoh/Config.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    configFile: String?,
+    connectEndpoints: List<String>?,
+    listenEndpoints: List<String>?,
+    noMulticastScouting: Boolean,
+    mode: String?
+): Config {
+    val config = if (emptyArgs) {
+        Config.default()
+    } else {
+        configFile?.let { Config.from(Path(it)) } ?: run {
```
Comment (oteffahi) [inline examples/src/main/kotlin/io.zenoh/Config.kt:63]: This seems to only support full path. Is it possible to add support for relative path?

Comment (DariusIMP) [reply examples/src/main/kotlin/io.zenoh/Config.kt:63]: Really? I test it with a relative path and it finds the file... Full path works as well.

Comment (DariusIMP) [reply examples/src/main/kotlin/io.zenoh/Config.kt:63]: As discussed internally, the issue is that the path is taken relative to where the examples's `build.gradle.kts` file is located. I consider it to be an out of scope issue for this PR, however I have opened issue #68 to tackle this in the future.

**Review by oteffahi — APPROVED** [review]
LGTM 👍

**Pull request merged**